test/integration/src/test/java/io/pravega/test/integration/KeyValueTableTest.java

```diff
@@ -192,6 +261,1427 @@ public void testCreateDeleteKeyValueTable() {
+        }
+
+        // Deleting KeyValueTable such that KVT does not exist within the Scope.
+        ResultObserver<DeleteKVTableStatus> result1 = new ResultObserver<>();
+        this.controllerService.deleteKeyValueTable(ModelHelper.createKeyValueTableInfo(newScope2,
+                newKVT1.getKeyValueTableName()), result1);
+        assertEquals("The requested table to delete must not be found for invalid Scope name",
+                DeleteKVTableStatus.Status.TABLE_NOT_FOUND, result1.get().getStatus());
+
+        // Delete KeyValueTable where KVTable does not exist within the Scope.
+        ResultObserver<DeleteKVTableStatus> result2 = new ResultObserver<>();
+        assertTrue("The requested Scope must get created", this.controller.createScope(newScope2).join());
+        this.controllerService.deleteKeyValueTable(ModelHelper.createKeyValueTableInfo(newScope1,
+                newKVT2.getKeyValueTableName()), result2);
+        assertEquals("The requested table to delete must not be found for invalid Table name",
+                DeleteKVTableStatus.Status.TABLE_NOT_FOUND, result2.get().getStatus());
+
+        final ScalingPolicy scalingPolicy = ScalingPolicy.fixed(5);
+        final StreamConfiguration strmConfig = StreamConfiguration.builder()
+                .scalingPolicy(scalingPolicy)
+                .build();
+
+        // Creating Stream1 under Scope1.
+        assertTrue("The requested Stream must get created", this.controller.createStream(newScope1,
+                streamName1, strmConfig).join());
+
+        val strmSegments1 = this.controller.getCurrentSegments(newScope1, streamName1).join();
+        for (val s : strmSegments1.getSegments()) {
+            this.tableStore.get(s.getScopedName(), Collections.singletonList(new ByteArraySegment(new byte[1])), TIMEOUT);
+        }
+
+        // KVTable and its segments should get deleted but Stream and its segments should exist and +
+        // we should be able to read/write from them.
+        ResultObserver<DeleteKVTableStatus> result3 = new ResultObserver<>();
+        this.controllerService.deleteKeyValueTable(ModelHelper.createKeyValueTableInfo(newScope1,
+                newKVT1.getKeyValueTableName()), result3);
+        assertEquals("The requested table to delete must not be found for invalid Scope name",
+                DeleteKVTableStatus.Status.SUCCESS, result3.get().getStatus());
+
+        for (val s : tableSegments1.getSegments()) {
+            AssertExtensions.assertSuppliedFutureThrows(
+                    "Segment " + s + " has not been deleted.",
+                    () -> this.tableStore.get(s.getKVTScopedName(), Collections.singletonList(new ByteArraySegment(new byte[1])), TIMEOUT),
+                    ex -> ex instanceof StreamSegmentNotExistsException);
+        }
+
+        for (val s : strmSegments1.getSegments()) {
+            this.tableStore.get(s.getScopedName(), Collections.singletonList(new ByteArraySegment(new byte[1])), TIMEOUT);
```

Comment:
Please fix this everywhere in your tests. You are not waiting on the returned Future to complete so this loop isn't providing any value.

Reply:
Sure, will fix this at all places. Thanks